AmplifyTests/CategoryTests/Auth/AuthCategoryConfigurationTests.swift

```diff
@@ -0,0 +1,367 @@
+    ///
+    func testCanAddPlugin() throws {
+        let plugin = MockAuthCategoryPlugin()
+        XCTAssertNoThrow(try Amplify.add(plugin: plugin))
+    }
+
+    /// Test if Auth plugin can be configured
+    ///
+    /// - Given: UnConfigured Amplify framework
+    /// - When:
+    ///    - I add a new Auth plugin and add configuration for the plugin
+    /// - Then:
+    ///    - Auth plugin should be configured correctly
+    ///
+    func testCanConfigureCategory() throws {
+        let plugin = MockAuthCategoryPlugin()
+        try Amplify.add(plugin: plugin)
+
+        let config = AuthCategoryConfiguration(
+            plugins: ["MockAuthCategoryPlugin": true]
+        )
+
+        let amplifyConfig = AmplifyConfiguration(auth: config)
+
+        try Amplify.configure(amplifyConfig)
+
+        XCTAssertNotNil(Amplify.Auth)
+        XCTAssertNotNil(try Amplify.Auth.getPlugin(for: "MockAuthCategoryPlugin"))
+    }
+
+    /// Test if resetting Auth category works
+    ///
+    /// - Given: Amplify framework configured with Auth plugin
+    /// - When:
+    ///    - I call Amplify.reset()
+    /// - Then:
+    ///    - The plugin should invoke the reset method.
+    ///
+    func testCanResetCategory() throws {
+        let plugin = MockAuthCategoryPlugin()
+        let resetWasInvoked = expectation(description: "reset() was invoked")
+        plugin.listeners.append { message in
+            if message == "reset" {
+                resetWasInvoked.fulfill()
+            }
+        }
+        try Amplify.add(plugin: plugin)
+
+        let config = AuthCategoryConfiguration(
+            plugins: ["MockAuthCategoryPlugin": true]
+        )
+
+        let amplifyConfig = AmplifyConfiguration(auth: config)
+
+        try Amplify.configure(amplifyConfig)
+        Amplify.reset()
+        waitForExpectations(timeout: 1.0)
+    }
+
+    /// Test whether calling reset removes the plugin added
+    ///
+    /// - Given: Amplify framework configured with Auth plugin
+    /// - When:
+    ///    - I call Amplify.reset
+    /// - Then:
+    ///    - Auth plugin should no longer work
+    ///
+    func testResetRemovesAddedPlugin() throws {
+        let plugin = MockAuthCategoryPlugin()
+        try Amplify.add(plugin: plugin)
+
+        let config = AuthCategoryConfiguration(
+            plugins: ["MockAuthCategoryPlugin": true]
+        )
+
+        let amplifyConfig = AmplifyConfiguration(auth: config)
+
+        try Amplify.configure(amplifyConfig)
+        Amplify.reset()
+        XCTAssertThrowsError(try Amplify.Auth.getPlugin(for: "MockAuthCategoryPlugin"),
+                             "Getting a plugin after reset() should throw") { error in
+                                guard case AmplifyAuthError.configuration = error else {
+                                    XCTFail("Expected PluginError.noSuchPlugin")
+                                    return
+                                }
```

Comment:
is there a case we are expecting?  can we assert this?

Reply:
ah.. sorry, somehow i misread this test when i first went through the pr. resolving.